### PR TITLE
Refactor browser window

### PR DESCRIPTION
**Pull request by timsueberkrueb**
See #10.

### CODE REVIEW

**Comment by plfiorini** [comment]
@tim-sueberkrueb It crashes with this error:

```
./liri-browser 
Using QtWebEngine
QQmlComponent: Created graphical object was not placed in the graphics scene.
[1106/123449:ERROR:browser_main_loop.cc(217)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
ATTENTION: default value of option force_s3tc_enable overridden by environment.
[1106/123449:ERROR:gl_surface_qt.cpp(187)] GLContextHelper::getXConfig() failed.
Segmentation fault (core dumped)
```

**Comment by plfiorini** [comment]
Looks like a problem in QtWebEngine, unfortunately I'm on Arch which doesn't have debuginfo.
All I can see is:

```
ATTENTION: default value of option force_s3tc_enable overridden by environment.
[1106/124239:ERROR:gl_surface_qt.cpp(187)] GLContextHelper::getXConfig() failed.
[New Thread 0x7fff88d46700 (LWP 25338)]
[New Thread 0x7fff63ffe700 (LWP 25339)]
[New Thread 0x7fff613c5700 (LWP 25340)]

Thread 21 "Chrome_InProcGp" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff8affd700 (LWP 25334)]
0x00007ffff2d948c9 in ?? () from /usr/lib/libQt5WebEngineCore.so.5
(gdb) bt
#0  0x00007ffff2d948c9 in  () at /usr/lib/libQt5WebEngineCore.so.5
#1  0x00007ffff2d953ff in  () at /usr/lib/libQt5WebEngineCore.so.5
#2  0x00007ffff3ff564f in  () at /usr/lib/libQt5WebEngineCore.so.5
#3  0x00007ffff4024fa2 in  () at /usr/lib/libQt5WebEngineCore.so.5
#4  0x00007ffff402335c in  () at /usr/lib/libQt5WebEngineCore.so.5
#5  0x00007ffff401ec3e in  () at /usr/lib/libQt5WebEngineCore.so.5
#6  0x00007ffff47c9789 in  () at /usr/lib/libQt5WebEngineCore.so.5
#7  0x00007ffff4772b58 in  () at /usr/lib/libQt5WebEngineCore.so.5
#8  0x00007ffff47737c9 in  () at /usr/lib/libQt5WebEngineCore.so.5
#9  0x00007ffff4773c92 in  () at /usr/lib/libQt5WebEngineCore.so.5
#10 0x00007ffff4775cb9 in  () at /usr/lib/libQt5WebEngineCore.so.5
#11 0x00007ffff4787f54 in  () at /usr/lib/libQt5WebEngineCore.so.5
#12 0x00007ffff4771ee5 in  () at /usr/lib/libQt5WebEngineCore.so.5
#13 0x00007ffff47a0d6a in  () at /usr/lib/libQt5WebEngineCore.so.5
#14 0x00007ffff479d0dc in  () at /usr/lib/libQt5WebEngineCore.so.5
#15 0x00007fffeed9f454 in start_thread () at /usr/lib/libpthread.so.0
#16 0x00007ffff0ade7df in clone () at /usr/lib/libc.so.6
```

File: src/ui/expansionbar/ExpansionBar.qml [hunk 1]
```diff
@@ -23,34 +23,78 @@
...
+Item {
     id: expansionBar
 
     property list<Component> contentComponents
     property int currentContentIndex: -1
+    property bool showing: false
 
-    signal closed()
+    signal opened();
+    signal closed();
```
Comment (plfiorini) [inline src/ui/expansionbar/ExpansionBar.qml:34]: code style: no semicolon here

File: src/ui/overlay/SearchOverlay.qml [hunk 2]
```diff
@@ -25,19 +25,29 @@ import QtQuick 2.7
...
-    id: searchBarItem
+BaseOverlay {
+    id: searchOverlay
 
     property bool searchEnabled: true
 
     signal searchRequested(string text, bool backwards)
 
+    onOpened: {
+        searchField.forceActiveFocus();
```
Comment (plfiorini) [inline src/ui/overlay/SearchOverlay.qml:37]: This doesn't work although the code looks OK

File: src/ui/overlay/SearchOverlay.qml [hunk 3]
```diff
@@ -25,19 +25,29 @@ import QtQuick 2.7
...
-        spacing: 8
+        anchors {
+            top: parent.top
+            bottom: parent.bottom
+        }
+
+        Item { implicitWidth: 16 }  // Spacer
 
         TextField {
             id: searchField
```
Comment (plfiorini) [inline src/ui/overlay/SearchOverlay.qml:53]: clicking on the textfield doesn't give it focus, not sure what's going on maybe an issue with the overlay?

File: src/ui/window/BrowserWindow.qml [hunk 4]
```diff
@@ -262,10 +226,9 @@ FluidWindow {
         }
 
         MenuItem {
-            text: "Search in page"
+            text: "Find in page"
```
Comment (plfiorini) [inline src/ui/window/BrowserWindow.qml:229]: use qsTr("Find in page") so in the future the string will be marked for translation

**Review by plfiorini — CHANGES_REQUESTED** [review]

File: src/ui/overlay/SearchOverlay.qml (hunk 3 above)
Comment (timsueberkrueb) [reply src/ui/overlay/SearchOverlay.qml:53]: @plfiorini it's disabled right now when no webpage is opened. Later we will ensure there's always a page opened or the menu item should probably be disabled.

File: src/ui/window/BrowserWindow.qml (hunk 4 above)
Comment (timsueberkrueb) [reply src/ui/window/BrowserWindow.qml:229]: @plfiorini Yeah, all strings are untranslatable right now, I will create a separate issue for that.

File: src/ui/overlay/SearchOverlay.qml (hunk 3 above)
Comment (plfiorini) [reply src/ui/overlay/SearchOverlay.qml:53]: @tim-sueberkrueb is it possible to make the menu item disabled with this PR?

Comment (timsueberkrueb) [reply src/ui/overlay/SearchOverlay.qml:53]: @plfiorini good idea

File: src/ui/window/BrowserWindow.qml (hunk 4 above)
Comment (plfiorini) [reply src/ui/window/BrowserWindow.qml:229]: @tim-sueberkrueb ok

Comment (timsueberkrueb) [reply src/ui/window/BrowserWindow.qml:229]: @plfiorini see #13.

File: src/ui/overlay/SearchOverlay.qml (hunk 2 above)
Comment (timsueberkrueb) [reply src/ui/overlay/SearchOverlay.qml:37]: It works when it's enabled.

**Review by plfiorini — APPROVED** [review]

**Pull request merged**